In [1]:
import pandas as pd 

In [2]:
df = pd.read_csv('../input/titanic/train.csv')
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
df.shape

(891, 12)

**Cleaning Data**

In [4]:
df = df.fillna(method = 'pad')
df = df.drop(columns = ['PassengerId','Name','Ticket','Cabin'])
df

,Survived,Pclass,Sex,Age,SibSp,Parch,Fare,Embarked
0,0,3,male,22.0,1,0,7.2500,S
1,1,1,female,38.0,1,0,71.2833,C
2,1,3,female,26.0,0,0,7.9250,S
3,1,1,female,35.0,1,0,53.1000,S
4,0,3,male,35.0,0,0,8.0500,S
...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,S
887,1,1,female,19.0,0,0,30.0000,S
888,0,3,female,19.0,1,2,23.4500,S
889,1,1,male,26.0,0,0,30.0000,C


**Convert Categorical Features or feature with string Value to Numerical Value,e.g: Sex and Embarked feature LabelEncoder Method**

In [5]:
from sklearn.preprocessing import LabelEncoder

**Create a model from LabelEncoder and fit_transform method such as le**

In [6]:
le = LabelEncoder()
df['Sex'] = le.fit_transform(df['Sex'])
df['Embarked'] = le.fit_transform(df['Embarked'])

**Target is Survived feature**

In [7]:
targets = df['Survived']
df = df.drop(columns = ['Survived'])
df

,Pclass,Sex,Age,SibSp,Parch,Fare,Embarked
0,3,1,22.0,1,0,7.2500,2
1,1,0,38.0,1,0,71.2833,0
2,3,0,26.0,0,0,7.9250,2
3,1,0,35.0,1,0,53.1000,2
4,3,1,35.0,0,0,8.0500,2
...,...,...,...,...,...,...,...
886,2,1,27.0,0,0,13.0000,2
887,1,0,19.0,0,0,30.0000,2
888,3,0,19.0,1,2,23.4500,2
889,1,1,26.0,0,0,30.0000,0


In [8]:
from sklearn.model_selection import train_test_split

**x_train is your Data**
**y_trani is your Labels or targets**

In [9]:
x_train,x_test,y_train,y_test = train_test_split(df,targets,test_size = 0.3)

**SVM = Support Vector Machine**

In [10]:
from sklearn import svm

**SVC = Support Vector Classification**

In [11]:
svm_a = svm.SVC()
svm_a.fit(x_train,y_train)

SVC()

**predict x_test data andobtain the accuracy of the SVM algorithm**

In [12]:
from sklearn import metrics
predicted = svm_a.predict(x_test)
print(metrics.classification_report(y_test,predicted))

              precision    recall  f1-score   support

           0       0.65      0.88      0.75       161
           1       0.62      0.29      0.39       107

    accuracy                           0.65       268
   macro avg       0.64      0.59      0.57       268
weighted avg       0.64      0.65      0.61       268



# accuracy of SVM algorithm is 0.67

**Train with decision tree algorithm**

In [13]:
from sklearn.tree import DecisionTreeClassifier
dt = DecisionTreeClassifier()
dt.fit(x_train,y_train)

DecisionTreeClassifier()

**predict x_test data andobtain the accuracy of the DecisionTreeClassifier algorithm**

In [14]:
predicted1 = dt.predict(x_test)
print(metrics.classification_report(y_test,predicted1))

              precision    recall  f1-score   support

           0       0.74      0.83      0.79       161
           1       0.69      0.57      0.63       107

    accuracy                           0.73       268
   macro avg       0.72      0.70      0.71       268
weighted avg       0.72      0.73      0.72       268



# accuracy of DecissionTree algorithm is 0.74

# Result: Dession Tree is better accuracy than SVM